app/src/shared/components/ElevatedContainer.jsx

```diff
@@ -0,0 +1,51 @@
+// @flow
+
+import React, { useState, useEffect } from 'react'
+import styled, { css } from 'styled-components'
+import { TOOLBAR_SHADOW, MD } from '$shared/utils/styled'
+import useInt from '$shared/hooks/useInt'
+
+export const Root = styled.div`
+    ${({ elevated, offset }) => (offset === 0 || !!elevated) && css`
+        box-shadow: ${TOOLBAR_SHADOW};
+        transition: 200ms box-shadow;
+    `}
+
+    ${({ elevated }) => !elevated && css`
+        @media (min-width: ${MD}px) {
```

Comment:
On desktop we enforce no-shadow if not elevated, obviously. This behaviour means that on small screens shadow is "always" there (see note about `offset`) and on desktop it shows up when necessary (not scrolled all the way up - offset).